mart/attack/adversary.py

```diff
@@ -172,8 +170,31 @@ def forward(self, *, model=None, sequence=None, **batch):
 
         return input_adv
 
-    def _attack(self, *, input, **batch):
+    # Make sure we can do autograd.
```

Comment:
An alternative is `+trainer.inference_mode=false` for experiments that run adversarial robustness evaluation in val/test.

Reply:
I don't like to turn off inference_mode for all experiments due to the performance advantage.

Reply:
I would make `trainer.inference_mode=false` in the default setup, because this is an adversary library. I also keep the `de_inference()` method because it can be used by applications that runs in the inference mode.

Reply:
I removed the `de_inference()` stuff that should live in a separate PR.